### PR TITLE
Silence known deprecation warnings

### PR DESCRIPTION
**Pull request by gnunicorn**
1. Prefixes known instances of usages of client.backend and client.import_lock with `#[allow(deprecated)]` to silence the warnings.
2. Remove file-global `#![allow(deprecated)]` used in these cases. Both to prevent us overlooking externally caused deprecation messages.

### CODE REVIEW

**Comment by ltfschoen** [comment]
@gnunicorn i was getting the following warnings when i ran `./scripts/build.sh && cargo build --release`

<details><p><pre>
   Compiling substrate-basic-authorship v2.0.0 (/home/ltfschoen/code/github/paritytech/substrate/core/basic-authorship)
warning: use of deprecated item '<substrate_client::client::Client<B, E, Block, RA>>::backend': Rather than relying on `client` to provide this, access to the backend should be handled at setup only - see #1134. This function will be removed once that is in place.
   --> core/network/src/chain.rs:137:9
    |
137 |             self.backend().blockchain(),
    |                  ^^^^^^^
    |
    = note: #[warn(deprecated)] on by default

warning: use of deprecated item '<substrate_client::client::Client<B, E, Block, RA>>::backend': Rather than relying on `client` to provide this, access to the backend should be handled at setup only - see #1134. This function will be removed once that is in place.
   --> core/network/src/chain.rs:137:9
    |
137 |             self.backend().blockchain(),
    |                  ^^^^^^^
    |
    = note: #[warn(deprecated)] on by default

   Compiling srml-consensus v2.0.0 (/home/ltfschoen/code/github/paritytech/substrate/srml/consensus)
   Compiling srml-timestamp v2.0.0 (/home/ltfschoen/code/github/paritytech/substrate/srml/timestamp)
   Compiling srml-democracy v2.0.0 (/home/ltfschoen/code/github/paritytech/substrate/srml/democracy)
   Compiling srml-balances v2.0.0 (/home/ltfschoen/code/github/paritytech/substrate/srml/balances)
   Compiling srml-finality-tracker v2.0.0 (/home/ltfschoen/code/github/paritytech/substrate/srml/finality-tracker)
   Compiling srml-indices v2.0.0 (/home/ltfschoen/code/github/paritytech/substrate/srml/indices)
   Compiling srml-executive v2.0.0 (/home/ltfschoen/code/github/paritytech/substrate/srml/executive)
   Compiling srml-sudo v2.0.0 (/home/ltfschoen/code/github/paritytech/substrate/srml/sudo)
   Compiling substrate-rpc v2.0.0 (/home/ltfschoen/code/github/paritytech/substrate/core/rpc)
warning: redundant linker flag specified for library `stdc++`

warning: use of deprecated item 'std::error::Error::cause': replaced by Error::source, which can support downcasting
  --> core/rpc/src/author/error.rs:26:1
   |
26 | / error_chain! {
27 | |     foreign_links {
28 | |         Client(client::error::Error) #[doc = "Client error"];
29 | |     }
...  |
49 | |     }
50 | | }
   | |_^
   |
   = note: #[warn(deprecated)] on by default
   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)

warning: use of deprecated item 'std::error::Error::cause': replaced by Error::source, which can support downcasting
  --> core/rpc/src/state/error.rs:22:1
   |
22 | / error_chain! {
23 | |     foreign_links {
24 | |         Client(client::error::Error) #[doc = "Client error"];
25 | |     }
...  |
38 | |     }
39 | | }
   | |_^
   |
   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)

warning: redundant linker flag specified for library `stdc++`

   Compiling srml-session v2.0.0 (/home/ltfschoen/code/github/paritytech/substrate/srml/session)
   Compiling srml-contract v2.0.0 (/home/ltfschoen/code/github/paritytech/substrate/srml/contract)
   Compiling rocksdb v0.11.0
warning: use of deprecated item 'std::error::Error::cause': replaced by Error::source, which can support downcasting
  --> core/rpc/src/author/error.rs:26:1
   |
26 | / error_chain! {
27 | |     foreign_links {
28 | |         Client(client::error::Error) #[doc = "Client error"];
29 | |     }
...  |
49 | |     }
50 | | }
   | |_^
   |
   = note: #[warn(deprecated)] on by default
   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)

warning: use of deprecated item 'std::error::Error::cause': replaced by Error::source, which can support downcasting
  --> core/rpc/src/state/error.rs:22:1
   |
22 | / error_chain! {
23 | |     foreign_links {
24 | |         Client(client::error::Error) #[doc = "Client error"];
25 | |     }
...  |
38 | |     }
39 | | }
   | |_^
   |
   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)

   Compiling substrate-rpc-servers v2.0.0 (/home/ltfschoen/code/github/paritytech/substrate/core/rpc-servers)
   Compiling kvdb-rocksdb v0.1.4 (https://github.com/paritytech/parity-common?rev=b0317f649ab2c665b7987b8475878fc4d2e1f81d#b0317f64)
   Compiling substrate-client-db v2.0.0 (/home/ltfschoen/code/github/paritytech/substrate/core/client/db)
   Compiling srml-staking v2.0.0 (/home/ltfschoen/code/github/paritytech/substrate/srml/staking)
warning: unnecessary path disambiguator
   --> srml/staking/src/lib.rs:399:27
    |
399 | type ExpoMap<T> = BTreeMap::<<T as system::Trait>::AccountId, Exposure<<T as system::Trait>::AccountId, BalanceOf<T>>>;
    |                           ^^ try removing `::`

   Compiling srml-grandpa v2.0.0 (/home/ltfschoen/code/github/paritytech/substrate/srml/grandpa)
   Compiling srml-treasury v2.0.0 (/home/ltfschoen/code/github/paritytech/substrate/srml/treasury)
   Compiling substrate-service v2.0.0 (/home/ltfschoen/code/github/paritytech/substrate/core/service)
   Compiling srml-council v2.0.0 (/home/ltfschoen/code/github/paritytech/substrate/srml/council)
warning: missing documentation for a variant
  --> core/service/src/error.rs:29:1
   |
29 | / error_chain! {
30 | |     foreign_links {
31 | |         Client(client::error::Error) #[doc="Client error"];
32 | |         Io(::std::io::Error) #[doc="IO error"];
...  |
46 | |     }
47 | | }
   | |_^
   |
note: lint level defined here
  --> core/service/src/lib.rs:20:9
   |
20 | #![warn(missing_docs)]
   |         ^^^^^^^^^^^^
   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)

   Compiling substrate-cli v2.0.0 (/home/ltfschoen/code/github/paritytech/substrate/core/cli)
warning: use of deprecated item '<substrate_client::client::Client<B, E, Block, RA>>::backend': Rather than relying on `client` to provide this, access to the backend should be handled at setup only - see #1134. This function will be removed once that is in place.
  --> core/cli/src/informant.rs:79:28
   |
79 |             let backend = (*client).backend();
   |                                     ^^^^^^^
   |
   = note: #[warn(deprecated)] on by default

warning: use of deprecated item '<substrate_client::client::Client<B, E, Block, RA>>::backend': Rather than relying on `client` to provide this, access to the backend should be handled at setup only - see #1134. This function will be removed once that is in place.
   --> core/cli/src/informant.rs:128:13
    |
128 |                     client.backend().blockchain(),
    |                            ^^^^^^^

   Compiling node-cli v2.0.0 (/home/ltfschoen/code/github/paritytech/substrate/node/cli)
warning: missing documentation for a variant
  --> core/service/src/error.rs:29:1
   |
29 | / error_chain! {
30 | |     foreign_links {
31 | |         Client(client::error::Error) #[doc="Client error"];
32 | |         Io(::std::io::Error) #[doc="IO error"];
...  |
46 | |     }
47 | | }
   | |_^
   |
note: lint level defined here
  --> core/service/src/lib.rs:20:9
   |
20 | #![warn(missing_docs)]
   |         ^^^^^^^^^^^^
   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)

   Compiling substrate-finality-grandpa v2.0.0 (/home/ltfschoen/code/github/paritytech/substrate/core/finality-grandpa)
warning: use of deprecated item '<substrate_client::client::Client<B, E, Block, RA>>::backend': Rather than relying on `client` to provide this, access to the backend should be handled at setup only - see #1134. This function will be removed once that is in place.
  --> core/cli/src/informant.rs:79:28
   |
79 |             let backend = (*client).backend();
   |                                     ^^^^^^^
   |
   = note: #[warn(deprecated)] on by default

warning: use of deprecated item '<substrate_client::client::Client<B, E, Block, RA>>::backend': Rather than relying on `client` to provide this, access to the backend should be handled at setup only - see #1134. This function will be removed once that is in place.
   --> core/cli/src/informant.rs:128:13
    |
128 |                     client.backend().blockchain(),
    |                            ^^^^^^^

</pre></p></details>

**Comment by ltfschoen** [comment]
@gnunicorn sorry, ignore my review comments, i didn't realise i ran it when i was still in the 'master' branch :-/

**Comment by gavofyork** [comment]
doesn't build 

```
error[E0658]: attributes on expressions are experimental. (see issue #15701)
    --> core/client/src/client.rs:2379:4
     |
2379 |             #[allow(deprecated)]
     |             ^^^^^^^^^^^^^^^^^^^^
```

File: core/consensus/aura/src/lib.rs [hunk 1]
```diff
@@ -1026,9 +1026,10 @@ mod tests {
 		let mut runtime = current_thread::Runtime::new().unwrap();
 		for (peer_id, key) in peers {
 			let client = net.lock().peer(*peer_id).client().as_full().expect("full clients are created").clone();
+			#[allow(deprecated)]
 			let select_chain = LongestChain::new(
 				client.backend().clone(),
-				client.import_lock().clone(),
+				client.import_lock().clone()
```
Comment (gnunicorn) [inline core/consensus/aura/src/lib.rs]:
```suggestion
				client.import_lock().clone(),
```

**Pull request merged**